### PR TITLE
Makes electrohack battery-powered

### PR DESCRIPTION
**Pull request by chaosvolt**
Idea that came up during electrohack-related discussion ( see https://github.com/CleverRaven/Cataclysm-DDA/pull/15103 in particular ). Potentially beefing up the fingerhack's power drain is also an option, but we'll see.
- Altered hack attempt to require charges from an electrohack, and drain 10 per hack attempt.
- Made electrohacks loadable with up to 100 batteries.

Gave it a compile-test and it seems to work fine, but will want to look at it further just to be sure.

### CODE REVIEW

**Comment by Core0verload** [comment]
How about adding electrohack function to control laptop, with maybe higher success rates?

**Comment by LlubNek** [comment]
I think the idea is that the electrohack is powered by the device you're hacking.  It should require electronics skill, computer skill and intelligence, not power.  It's runs more on caffeine than batteries.

File: src/iexamine.cpp [hunk 1]
```diff
@@ -2969,7 +2969,7 @@ void iexamine::pay_gas(player *p, map *m, const tripoint &examp)
     int pricePerUnit = getPricePerGasUnit(discount);
     std::string unitPriceStr = string_format(_("$%0.2f"), pricePerUnit / 100.0);
 
-    bool can_hack = (!p->has_trait("ILLITERATE") && ((p->has_amount("electrohack", 1)) ||
+    bool can_hack = (!p->has_trait("ILLITERATE") && ((p->has_charges("electrohack", 10)) ||
                      (p->has_bionic("bio_fingerhack") && p->power_level > 0)));
```
Comment (Coolthulhu) [inline src/iexamine.cpp]: Not really related to electrohack, but here the fingerhack should require 25 energy

**Comment by Coolthulhu** [comment]
Electrohack is rather tiny, it shouldn't be more efficient than fingerhack.
25 energy per use would be better.
Even better would be letting it use some non-hardcoded value and just set it in the json.

**Comment by John-Candlebury** [comment]
>How about adding electrohack function to control laptop, with maybe higher success rates?

I like the idea of adding the electrohack function to the control laptop.

In my ideal world the laptop would be the general do it everything tool for hacking.

**Comment by chaosvolt** [comment]
> I think the idea is that the electrohack is powered by the device you're hacking. It should require electronics skill, computer skill and intelligence, not power. It's runs more on caffeine than batteries.

I had assumed that was the reasoning behind electrohacks not using batteries, yeah. This is more to be consistent with the fact that fingerhack also doesn't rely on power from the target device (though I suppose a concealed fingerhack might need power to open up).

> Electrohack is rather tiny, it shouldn't be more efficient than fingerhack.
25 energy per use would be better.
Even better would be letting it use some non-hardcoded value and just set it in the json.

I was thinking I might go with 25 energy for it, sure. As for JSONized power, I'm unsure how to go about that. o.o

In any case, heading out again, will be able to mess with this when I get back. @_@

**Comment by chaosvolt** [comment]
There we go. Still unsure how to go about JSONizing it.

**Comment by chaosvolt** [comment]
And thank you for the merge.

**Pull request merged**